backend/dal/Services/Admin/Concrete/AgencyService.cs

```diff
@@ -207,7 +207,7 @@ public override void Remove(Agency agency)
         {
             agency.ThrowIfNull(nameof(agency));
 
-            var original = this.Context.Agencies.Find(agency.Id) ?? throw new KeyNotFoundException();
+            var original = this.Context.Agencies.Include(a => a.Users).FirstOrDefault(a => a.Id == agency.Id) ?? throw new KeyNotFoundException();
```

Comment:
not including the useragencies causes the cascade to fail.

Reply:
The downside to this is `include(...)` statements are expensive operations.

Reply:
this should be fine since it is finding by an indexed id.